desktop/core/src/desktop/auth/api_authentications.py

```diff
@@ -49,10 +50,15 @@ def authenticate(self, request):
 
     LOG.debug('JwtAuthentication: got access token %s' % access_token)
 
+    # Fetch public key
+    token_metadata = jwt.get_unverified_header(access_token)
+    headers = {'kid': token_metadata.get('kid')} if token_metadata.get('kid') is not None else {}
```

Comment:
Often providing a default value for the get is simpler:

`token_metadata.get('kid', {})`